### PR TITLE
Policy Mechanism

### PR DESCRIPTION
**Pull request by chaen**
Linked to https://github.com/DIRACGrid/diracx/issues/161

needs https://github.com/tiangolo/fastapi/pull/11355

### CODE REVIEW

File: diracx-routers/src/diracx/routers/__init__.py [hunk 1]
```diff
@@ -1,3 +1,14 @@
+"""
+# Startup sequence
+
+uvicorn is called with `create_app` as a factory
+
+create_app loads the environment configuration
+
+
+
```
Comment (chrisburr) [inline diracx-routers/src/diracx/routers/__init__.py:9]:
```suggestion
```

File: diracx-routers/src/diracx/routers/auth/access_policies.py
Comment (chrisburr) [inline diracx-routers/src/diracx/routers/auth/access_policies.py:1]: Junk?

File: diracx-routers/src/diracx/routers/utils/users.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    # TODO: This should be taken dynamically
+    # ["http://pclhcb211:8000/docs/oauth2-redirect"]
+    allowed_redirects: list[str] = []
+    device_flow_expiration_seconds: int = 600
+    authorization_flow_expiration_seconds: int = 300
+
+    # State key is used to encrypt/decrypt the state dict passed to the IAM
+    state_key: FernetKey
+
+    token_issuer: str = "http://lhcbdirac.cern.ch/"
```
Comment (chrisburr) [inline diracx-routers/src/diracx/routers/utils/users.py:50]: I guess this should be mandatory? If not, we should at least have:

```suggestion
    token_issuer: str = "https://my-issuer.invalid/"
```

**Review by chrisburr — COMMENTED** [review]
Looks good in general, just a couple of minor things and this needs rebasing.

File: diracx-routers/src/diracx/routers/utils/users.py (hunk 2 above)
Comment (chaen) [reply diracx-routers/src/diracx/routers/utils/users.py:52]: added the todo

**Review by chrisburr — APPROVED** [review]

**Pull request merged**